src/Build/Definition/ProjectCollection.cs

```diff
@@ -621,18 +610,20 @@ public bool IsBuildEnabled
             [DebuggerStepThrough]
             set
             {
-                ProjectCollectionChangedEventArgs eventArgs = null;
+                bool sendEvent = false;
                 using (_locker.EnterWriteLock())
                 {
                     if (_isBuildEnabled != value)
                     {
                         _isBuildEnabled = value;
-
-                        eventArgs = new ProjectCollectionChangedEventArgs(ProjectCollectionChangedState.IsBuildEnabled);
+                        sendEvent = true;
                     }
                 }
 
-                OnProjectCollectionChangedIfNonNull(eventArgs);
+                if (sendEvent)
+                {
+                    OnProjectCollectionChanged(new ProjectCollectionChangedEventArgs(ProjectCollectionChangedState.IsBuildEnabled));
```

Comment:
`OnProjectCollectionChangedIfNonNull` seems like it can be deleted now.

Reply:
true